### ISSUE TITLE
Limit brokered SolrCloud service endpoints to cloud.gov egress IP ranges

### ISSUE THREAD

**Issue by mogul** [GSA/data.gov#2971]
## User Story

In order to minimize the public surface area of back-end services, we want SSB-brokered SolrCloud instances to limit client IP ranges to [the cloud.gov egress ranges](https://cloud.gov/docs/management/static-egress/#cloudgov-egress-ranges) by default. 

## Acceptance Criteria
[ACs should be clearly demoable/verifiable whenever possible. Try specifying them using [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development#Behavioral_specifications).]
- [ ] GIVEN I have provisioned an instance of SolrCloud \
  AND I am running commands **outside** of cloud.gov (eg on my workstation) \
  WHEN I run `curl ...` to the broker-provided URL with the broker-provided credentials \
  THEN the command times out with no response.
- [ ] GIVEN I have provisioned an instance of SolrCloud \
  AND I am running commands **inside** of cloud.gov (eg after running `cf ssh`) \
  WHEN I run `curl ...` to the broker-provided URL with the broker-provided credentials \
  THEN I see a 200 response.

## Background
[Any helpful contextual notes or links to artifacts/evidence, if needed]
Note we are explicitly NOT allowing for access to SolrCloud instances from outside the cloud.gov egress ranges here. At the conclusion of this issue, using a brokered SolrCloud should be just like using a brokered RDS instance... The credentials are only useful to clients behind the cloud.gov ranges.

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
This change will reduce the public-facing surface area of provisioned SolrCloud instances. This change mitigates the potential risk of credentials for a SolrCloud instance being exported from cloud.gov; they won't be useful from anywhere else.

## Sketch
[Notes or a checklist reflecting our understanding of the selected approach]

- Use the configuration enabled by #3355 to limit inbound traffic to cloud.gov egress ranges on the `ssb-solr-k8s` instance that backs `ssb-solr` (this should be the default once that issue is done, making support of the ACs nearly a no-op).
- Alternatively, IF #3665 is done, we can be more flexible... Set `ssb-solr-eks` to accept traffic from 0.0.0.0/0, and have the ssb-solrcloud broker specify a [NetworkPolicy](https://kubernetes.io/docs/concepts/services-networking/network-policies/) that accepts ingress traffic from cloud.gov egress IP ranges to the SolrCloud pods.
  - Out of scope for the ACs, but a nice enhancement: Allow specifying a custom set of client IP ranges to put in the SolrCloud NetworkPolicy at service-provisioning time.

**Comment by 0937145936** [comment]
thongmoongankham123456789@gmail.gov
